migrate/20250130_create_host_provider_table.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  up do
+    create_table(:host_provider) do
+      foreign_key :id, :vm_host, type: :uuid
+      column :server_identifier, :text, null: false, unique: true
+      column :provider_name, :text, null: false
+      index [:provider_name, :server_identifier], unique: true
```

Comment:
```suggestion
      column :server_identifier, :text
      column :provider_name, :text
      primary_key [:provider_name, :server_identifier]
```

It's best it all tables have a primary key.  This removes the unique constraint on just server_identifier, since I think it's possible for separate providers to use the same server_identifier, even if it is very unlikely.